src/pybind/mgr/cephadm/serve.py

```diff
@@ -520,6 +512,7 @@ def _apply_all_services(self) -> bool:
                                             len(self.mgr.apply_spec_fails),
                                             warnings)
         self.mgr.update_watched_hosts()
+        self._write_all_client_files()
```

Comment:
hmm, I'm wondering if the actual cause was calculating the list of client files before the host refresh had actually occurred and we should still write the client files during `_refresh_hosts_and_daemons` rather than much later in `_apply_all_services`?

Reply:
not sure, the calculated client files are calculated from just doing placements on the client file specs we have in the cache. That was the part that was being left blank. Would the hosts not being refreshed yet cause those placements to not come up with anything? I can try still doing this as part of `_refresh_hosts_and_daemons` but not calculating the client files until the hosts have been refreshed and see how the results compare.

Reply:
moved the call to `_write_all_client_files` to `_refresh_hosts_and_daemons` right after the host refresh. I think that's the earliest we can put this where it might still make a difference (i.e. calculating the client files after the hosts are refreshed) so we'll see how it goes.